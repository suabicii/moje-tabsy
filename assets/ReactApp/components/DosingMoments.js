import React from "react";

function DosingMoments({drugId, content}) {
    const dosingMomentsToDisplay = [];
    for (const value in content) {
        if (content.hasOwnProperty(value)) {
            const [hour, minute] = content[value].split(':');
            dosingMomentsToDisplay.push(
                <li key={drugId + value}>{hour}:{minute}</li>
            );
        }
    }
    return (
        <ul>
            {dosingMomentsToDisplay}
        </ul>
    );
}

export default DosingMoments;