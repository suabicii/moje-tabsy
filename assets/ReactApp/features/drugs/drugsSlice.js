import {createSelector, createSlice} from "@reduxjs/toolkit";
import {fetchData} from "../../utils/fetchData";

const drugsSlice = createSlice({
    name: 'drugs',
    initialState: [],
    reducers: {
        addDrug: (state, action) => {
            const {id, name, quantity, quantityMax, unit, dosing, dosingMoments} = action.payload;
            state.push({id, name, quantity, quantityMax, unit, dosing, dosingMoments});
        },
        removeDrug: (state, action) => {
            const id = action.payload;
            return state.filter(drug => drug.id !== id);
        },
        editDrug: (state, action) => {
            const {id, ...updates} = action.payload;
            const drugIndex = state.findIndex(drug => drug.id === id);
            if (drugIndex !== -1) {
                state[drugIndex] = {...state[drugIndex], ...updates};
            }
        },
        setDrugs: (state, action) => action.payload
    },
    sort: true
});

const {reducer, actions} = drugsSlice;
export const {addDrug, removeDrug, editDrug, setDrugs} = actions;

const defaultDrugsSelector = state => state.drugs;

export const sortedDrugsSelector = createSelector(
    defaultDrugsSelector,
    drugs => [...drugs].sort((a, b) => a.name.localeCompare(b.name))
);

export const fetchDrugs = () => async dispatch => {
    await fetchData('/api/drug-list').then(data => {
        dispatch(setDrugs(data));
    }).catch(err => {
        console.log(err);
    });
};

export default reducer;