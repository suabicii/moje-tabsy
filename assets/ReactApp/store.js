import {configureStore} from "@reduxjs/toolkit";
import drugsSlice from "./features/drugs/drugsSlice"
import userSlice from "./features/user/userSlice";
import darkModeSlice from "./features/darkMode/darkModeSlice";

export default configureStore({
    reducer: {
        drugs: drugsSlice,
        user: userSlice,
        darkMode: darkModeSlice
    }
});