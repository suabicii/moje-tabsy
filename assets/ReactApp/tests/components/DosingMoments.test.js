/**
 * @jest-environment jsdom
 * */
import React from "react";
import ReactShallowRenderer from "react-test-renderer/shallow";
import {Provider} from "react-redux";
import {render, screen} from "@testing-library/react";
import store from "../../store";
import DosingMoments from "../../components/DosingMoments";
import drugs from "./fixtures/drugs";

const drug = drugs[0];
const content = drug.dosingMoments;

it('should properly render DosingMoments component', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(
        <Provider store={store}>
            <DosingMoments content={content} drugId={drug.id}/>
        </Provider>
    );

    expect(renderer.getRenderOutput()).toMatchSnapshot();
});

it.each`
    hour
    ${Object.values(content)[0]}
    ${Object.values(content)[1]}
`('should display proper hours of dosing moments', ({hour}) => {
    render(
        <Provider store={store}>
            <DosingMoments content={content} drugId={drug.id}/>
        </Provider>
    );

    expect(screen.queryByText(hour)).not.toBeFalsy();
});