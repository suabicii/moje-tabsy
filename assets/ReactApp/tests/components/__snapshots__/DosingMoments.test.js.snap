// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should properly render DosingMoments component 1`] = `
<Context.Provider
  value={
    Object {
      "getServerState": undefined,
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <DosingMoments
    content={
      Object {
        "hour1": "07:00",
        "hour2": "18:00",
      }
    }
    drugId={1}
  />
</Context.Provider>
`;
